core/dbt/version.py

```diff
@@ -130,7 +130,11 @@ def _get_plugin_msg_info(
 
     needs_update = False
 
-    if plugin.major != core.major or plugin.minor != core.minor:
+    assert plugin.major is not None
```

Comment:
`asserts` shouldn't be used in production, see: https://snyk.io/blog/the-dangers-of-assert-in-python/